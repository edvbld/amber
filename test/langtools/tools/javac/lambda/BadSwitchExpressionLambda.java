/*
 * @test /nodynamiccopyright/
 * @bug 8206986
 * @summary Adding switch expressions
 * @compile/fail/ref=BadSwitchExpressionLambda.out -XDrawDiagnostics --enable-preview -source 12 BadSwitchExpressionLambda.java
 */

class BadSwitchExpressionLambda {

    interface SAM {
        void invoke();
    }

    public static void m() {}

    void test(int i) {
        SAM sam1 = () -> m(); //ok
        SAM sam2 = () -> switch (i) { case 0 -> m(); default -> m(); }; //not ok
    }
}
