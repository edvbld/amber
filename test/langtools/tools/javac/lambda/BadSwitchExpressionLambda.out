BadSwitchExpressionLambda.java:18:26: compiler.err.prob.found.req: (compiler.misc.incompatible.ret.type.in.lambda: (compiler.misc.switch.expression.target.cant.be.void))
- compiler.note.preview.filename: BadSwitchExpressionLambda.java
- compiler.note.preview.recompile
1 error
